Core/Core/MobileCore.cs

```diff
@@ -138,7 +142,13 @@ private MobileCore(IPlatformInjector injector, Options options)
         public static MobileCore Init(IPlatformInjector injector, Options options)
         {
             NonNull<Options>(options, "init options");
-            instance = new MobileCore(injector, options);
+            lock (coreLock)
```

Comment:
any particular reason we need to lock here?

Reply:
Because initializing the singleton should be synchronized so we don't have multiple instances.